src/megaclient.cpp

```diff
@@ -3582,23 +3576,11 @@ void MegaClient::freeq(direction_t d)
 }
 
 // determine next scheduled transfer retry
-// FIXME: make this an ordered set and only check the first element instead of
-// scanning the full map!
 void MegaClient::nexttransferretry(direction_t d, dstime* dsmin)
 {
-    for (transfer_map::iterator it = transfers[d].begin(); it != transfers[d].end(); it++)
+    if (!xferpaused[d])   // avoid setting the timer's next=1 if it won't be processed
```

Comment:
If some transfers are paused and we avoid to update its backoff timer, do we arm the timer when they are un-paused?

Reply:
Those two actually operate independently - xferpaused just corresponds to whether the user has clicked 'pause' for puts/gets.  If the timer was in a backoff state when those are unpaused, then we'll start evaluating it again, and process it if the specifed backoff time has been reached.   This check was needed specifically because the timers have a built in check that we are using them correctly, if the backoff is expired when we call preparewait(), it sets its internal flag next=1, expecting that the next exec() will do something with that backed-off object.   If it doesn't, then we get an assertion on the next preparewait().   So with the pause turned on, we would not make that adjustment in exec(), and so we need to avoid considering those backofftimers in preparewait().  cheers